pandasai/__init__.py

```diff
@@ -323,9 +323,14 @@ def _is_unsafe_import(self, node: ast.stmt) -> bool:
                 if alias.name in WHITELISTED_OPTIONAL_LIBRARIES:
                     import_optional_dependency(alias.name)
                     continue
-                if not any(
-                    alias.name.startswith(whitelisted_lib)
-                    for whitelisted_lib in WHITELISTED_LIBRARIES
+                if (
+                    alias.name not in WHITELISTED_LIBRARIES
+                    and alias.name != "pandas"
```

Comment:
This is the same as adding `pandas` to `WHITELISTED_LIBRARIES`. So the line `import pandas as pd` persists and will be included in the run code. `_is_unsafe_import()` should return `True` for imports of pandas.

Reply:
Here's a test:
```python
    def test_clean_code_remove_pandas(self, pandasai):
        pandas_code = """
import pandas as pd
print(pd.DataFrame())
"""
        pandasai._llm._output = pandas_code
        assert pandasai.run_code(pandas_code, pd.DataFrame()) == ""
        assert pandasai.last_run_code == "print(pd.DataFrame())"
```

Reply:
ok yes, we should return. I'll refactor

Reply:
Sorry, my test was bad, here's a better one:
```python
    def test_clean_code_remove_pandas(self, pandasai):
        pandas_code = """
import pandas as pd
print(df.size)
"""
        pandasai._llm._output = pandas_code
        pandasai.run_code(pandas_code, pd.DataFrame())
        assert pandasai.last_run_code == "print(df.size)"
```